apps/emqx_dashboard/src/emqx_dashboard.erl

```diff
@@ -61,15 +61,26 @@ start_listeners() ->
         dispatch => Dispatch,
         middlewares => [cowboy_router, ?EMQX_MIDDLE, cowboy_handler]
     },
-    [begin
-        Minirest = maps:put(protocol, Protocol, BaseMinirest),
-        {ok, _} = minirest:start(Name, RanchOptions, Minirest),
-        ?ULOG("Start listener ~ts on ~p successfully.~n", [Name, Port])
-    end || {Name, Protocol, Port, RanchOptions} <- listeners()].
+    Res =
+        lists:foldl(fun({Name, Protocol, Port, RanchOptions}, Acc) ->
+            Minirest = BaseMinirest#{protocol => Protocol},
+            case minirest:start(Name, RanchOptions, Minirest) of
+                {ok, _} ->
+                    ?ULOG("Start listener ~ts on ~p successfully.~n", [Name, Port]),
+                    Acc;
+                {error, _Reason} ->
+                    %% Don't record the reason because minirest already does(too much logs noise).
+                    [Name | Acc]
+            end
+                    end, [], listeners()),
+    case Res of
+        [] -> ok;
+        _ -> {error, Res}
+    end.
 
 stop_listeners() ->
     [begin
-        ok = minirest:stop(Name),
+        _ = minirest:stop(Name),
```

Comment:
I assume this means we don't care whether or not there is actually a listener to stop? Would it be worth it to have a case here that logs the fact that there was no listener with 'Name' found?